src/Illuminate/Queue/SerializesModels.php

```diff
@@ -11,67 +10,109 @@ trait SerializesModels
     use SerializesAndRestoresModelIdentifiers;
 
     /**
-     * The list of serialized model identifiers.
+     * Prepare the instance for serialization.
      *
-     * @var array
+     * @return array
      */
-    protected $modelIdentifiers = [];
+    public function __sleep()
+    {
+        $properties = (new ReflectionClass($this))->getProperties();
+
+        foreach ($properties as $property) {
+            $property->setValue($this, $this->getSerializedPropertyValue(
+                $this->getPropertyValue($property)
+            ));
+        }
+
+        return array_values(array_filter(array_map(function ($p) {
+            return $p->isStatic() ? null : $p->getName();
+        }, $properties)));
+    }
 
     /**
-     * Prepare the instance for serialization.
+     * Restore the model after serialization.
+     *
+     * @return void
+     */
+    public function __wakeup()
+    {
+        foreach ((new ReflectionClass($this))->getProperties() as $property) {
+            if ($property->isStatic()) {
+                continue;
+            }
+
+            $property->setValue($this, $this->getRestoredPropertyValue(
+                $this->getPropertyValue($property)
+            ));
+        }
+    }
+
+    /**
+     * Prepare the instance values for serialization.
      *
      * @return array
      */
-    public function __sleep()
+    public function __serialize()
     {
+        $values = [];
+
         $properties = (new ReflectionClass($this))->getProperties();
 
+        $class = get_class($this);
+
         foreach ($properties as $property) {
-            if ($property->getName() === 'modelIdentifiers') {
+            if ($property->isStatic()) {
                 continue;
             }
 
-            $serializedValue = $this->getSerializedPropertyValue(
-                $value = $this->getPropertyValue($property)
-            );
+            $name = $property->getName();
 
-            if ($serializedValue instanceof ModelIdentifier) {
-                $this->modelIdentifiers[$property->getName()] = $serializedValue;
-
-                // Empty instance of the model or collection to support typed properties...
-                $property->setValue($this, new $value);
-            } else {
-                $property->setValue($this, $value);
+            if ($property->isPrivate()) {
+                $name = "\0{$class}\0{$name}";
+            } elseif ($property->isProtected()) {
+                $name = "\0*\0{$name}";
```

Comment:
What is this stuff?

Reply:
PHP adds `\0*\0` for protected and `\0className\0` for private properties in serialization.
[serialization_objects_test](https://github.com/php/php-src/blob/d679f02295ea079338f029b7f5f4cb65b37f190c/ext/standard/tests/serialize/serialization_objects_001.phpt), [zend_mangle_property_name](https://github.com/php/php-src/blob/36afe4e39ec724eab19c54c1492db030721d0ec1/Zend/zend_compile.c#L1338) and usage [1](https://github.com/php/php-src/blob/647b1c7fcf04447658459f507271c0a5ce75229a/ext/standard/var.c#L825), [2](https://github.com/php/php-src/blob/647b1c7fcf04447658459f507271c0a5ce75229a/ext/standard/var.c#L844)

Reply:
It's needed for backward compatibility with php7.3 workers.

A php7.3 serialize command will produce this name in the serialized array so by naming it this way a 7.3 worker can work together with a 7.4 worker (as long as it's not using typed properties).